### PR TITLE
Make sure that the relevancy is not impacted by other settings

### PR DESCRIPTION
**Pull request by Kerollmops**
Fix https://github.com/meilisearch/meilisearch/issues/1505.

fix https://github.com/meilisearch/MeiliSearch/issues/1529

### CODE REVIEW

File: milli/src/search/mod.rs [hunk 1]
```diff
@@ -18,7 +18,6 @@ pub(crate) use self::facet::ParserRule;
 pub use self::facet::{FacetDistribution, FacetNumberIter, FilterCondition, Operator};
 pub use self::matching_words::MatchingWords;
 use self::query_tree::QueryTreeBuilder;
-use crate::error::FieldIdMapMissingEntry;
```
Comment (Kerollmops) [inline milli/src/search/mod.rs:21]: @MarinPostma  Is this error used anywhere else now? Can't we just remove it?

File: milli/src/update/index_documents/store.rs [hunk 2]
```diff
@@ -688,13 +688,13 @@ impl<'s, A: AsRef<[u8]>> Store<'s, A> {
                 }
 
                 for (attr, content) in document.iter() {
-                    if self.filterable_fields.contains(&attr)
-                        || self.searchable_fields.contains(&attr)
-                    {
+                    let is_filterable = self.filterable_fields.contains(dbg!(&attr));
```
Comment (Kerollmops) [inline milli/src/update/index_documents/store.rs:691]:
```suggestion
                    let is_filterable = self.filterable_fields.contains(&attr);
```

File: milli/src/update/index_documents/store.rs [hunk 3]
```diff
@@ -688,13 +688,13 @@ impl<'s, A: AsRef<[u8]>> Store<'s, A> {
...
+                    let is_searchable = self.searchable_fields.contains(&attr);
+                    if is_filterable || is_searchable {
```
Comment (Kerollmops) [inline milli/src/update/index_documents/store.rs:693]: @MarinPostma Could you revert this, please ?

File: milli/src/update/index_documents/store.rs (hunk 2 above)
Comment (Kerollmops) [inline milli/src/update/index_documents/store.rs:691]:
```suggestion
                    let is_filterable = self.filterable_fields.contains(&attr);
```

File: milli/src/update/settings.rs [hunk 4]
```diff
@@ -692,7 +674,8 @@ mod tests {
         let count = index
             .facet_id_f64_docids
             .remap_key_type::<ByteSlice>()
-            .prefix_iter(&rtxn, &[0, 0])
+            // The facetted field id is 2u16
```
Comment (Kerollmops) [inline milli/src/update/settings.rs:677]:
```suggestion
            // The faceted field id is 2u16
```

Comment (MarinPostma) [reply milli/src/update/settings.rs:677]: I mean that it is 2u16, not two u16 😅

File: milli/src/search/mod.rs (hunk 1 above)
Comment (MarinPostma) [reply milli/src/search/mod.rs:21]: this is used when building documents in obkv_json, at least

File: milli/src/update/index_documents/store.rs (hunk 2 above)
Comment (MarinPostma) [reply milli/src/update/index_documents/store.rs:691]: weird, but alright...

**Review by MarinPostma — APPROVED** [review]

**Comment by Kerollmops** [comment]
bors merge

**Review by MarinPostma — APPROVED** [review]

**Comment by Kerollmops** [comment]
bors merge

**Pull request merged**